File: Structural/Bridge/Cpp/AdvancedRemote.cpp
// AdvancedRemote.cpp

#include "AdvancedRemote.h"
#include <iostream>

AdvancedRemote::AdvancedRemote(IDevice* device) : BasicRemote(device) {}

void AdvancedRemote::mute() {
  device->setVolume(0);
  std::cout << "Device is muted." << std::endl;
}

File: Structural/Bridge/Java/AdvancedRemote.java
// AdvancedRemote.java

public class AdvancedRemote extends BasicRemote {
  public AdvancedRemote(Device device) {
    super(device);
  }

  public void mute() {
    device.setVolume(0);
    System.out.println("Device is muted.");
  }
}
